FantasmoSDK/Classes/ValueTypes/ARKitTrackingStateStatistics.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  FrameStatistics.swift
+//  FantasmoSDK
+//
+//  Created by Yauheni Klishevich on 13.07.2021.
+//
+
+import ARKit
+
+/// Statistics on frames by the quality of ARKit posion tracking.
+struct ARKitTrackingStateStatistics {
+    
+    /// Number of frames captured at the moment when tracking state was `ARFrame.camera.trackingState == .notAvailable`
+    private(set) var framesWithNotAvailableTracking: Int = 0
+    
+    /// Number of frames captured at the moment when tracking state was `ARFrame.camera.trackingState == .limited`
+    private(set) var framesWithLimitedTrackingStateByReason: [ARCamera.TrackingState.Reason : Int] = [
+        .initializing: 0,
+        .relocalizing: 0,
+        .excessiveMotion : 0,
+        .insufficientFeatures : 0,
+    ]
+
+    /// Number of frames captured at the moment when tracking state  was`ARFrame.camera.trackingState == .normal`
+    private(set) var framesWithNormalTrackingState: Int = 0
+    
+    private(set) var totalNumberOfFrames: Int = 0
+    
+    mutating func update(with trackingState: ARCamera.TrackingState) {
+        totalNumberOfFrames += 1
+        
+        switch trackingState {
+        case .limited(let reason):
+            framesWithLimitedTrackingStateByReason[reason]! += 1
```

Comment:
Can we get rid of the forced unwrap?

Reply:
I guess you propose to write:

`framesWithLimitedTrackingStateByReason[reason] = (framesWithLimitedTrackingStateByReason[reason] ?? 0) + 1`

and this will work stably in any circumstances (whether `framesWithLimitedTrackingStateByReason` is prepopulated with 0s or not).

Ok, agree